files/sdw-admin.py

```diff
@@ -270,6 +288,185 @@ def perform_uninstall():
     )
 
 
+def extract_fingerprint(gpg_output):
+    lines = gpg_output.strip().split("\n")
+
+    primary_key = True
+    for line in lines:
+        if not line.strip():
+            continue
+        fields = line.split(":")
+        record_type = fields[0]
+
+        if record_type == "sec":
+            primary_key = True
+            continue
+        if record_type == "ssb":
+            primary_key = False
+            continue
+        if record_type == "fpr" and primary_key and len(fields) > 9 and fields[9]:
+            return fields[9]
+    return None
+
+
+def _try_read_submission_key():
+    """
+    Checks if SecureDrop submission key is written to dom0. If so, returns
+    submission key fingerprint
+    """
+    if not os.path.exists(SCRIPTS_PATH + SUBMISSION_KEY):
+        return None
+    gpg_output = subprocess.check_output(
+        ["gpg", "--show-keys", "--with-fingerprint", "--with-colon", SCRIPTS_PATH + SUBMISSION_KEY],
+        text=True,
+    )
+    return extract_fingerprint(gpg_output)
+
+
+def import_submission_key():
```

Comment:
This assumes there's a single secret key in the keyring - there may be multiple ones if the submission key has been rotated previously. For now we just support using one key during provisioning, so if there are multiple keys there are a couple of possible approaches:
- (preferred) list the secret key user ids and fingerprints found, prompt the user to choose one (either via a numerical list or by typing the full fingerprint)
- exit with an error, saying multiple submission keys are not supported at installation
- something else?

Reply:
Great callout! Changed to list the secret key fingerprints + prompt the user to choose if there are multiple keys in the keyring